step/step.go

```diff
@@ -42,40 +51,68 @@ func New(
 	pathProvider pathutil.PathProvider,
 	pathModifier pathutil.PathModifier,
 	envRepo env.Repository,
+	derivedDataPathProvider xcodecache.SwiftPackageCache,
 ) SaveCacheStep {
 	return SaveCacheStep{
-		logger:       logger,
-		inputParser:  inputParser,
-		pathChecker:  pathChecker,
-		pathProvider: pathProvider,
-		pathModifier: pathModifier,
-		envRepo:      envRepo,
+		logger:                  logger,
+		inputParser:             inputParser,
+		pathChecker:             pathChecker,
+		pathProvider:            pathProvider,
+		pathModifier:            pathModifier,
+		envRepo:                 envRepo,
+		derivedDataPathProvider: derivedDataPathProvider,
 	}
 }
 
-func (step SaveCacheStep) Run() error {
+func (step SaveCacheStep) ProcessConfig() (Config, error) {
 	var input Input
 	if err := step.inputParser.Parse(&input); err != nil {
-		return fmt.Errorf("failed to parse inputs: %w", err)
+		return Config{}, err
 	}
 	stepconf.Print(input)
+	step.logger.EnableDebugLog(input.Verbose)
+
+	input.DerivedDataPath = strings.TrimSpace(input.DerivedDataPath)
+	input.ProjectPath = strings.TrimSpace(input.ProjectPath)
+	if input.DerivedDataPath == "" && input.ProjectPath == "" {
+		return Config{}, fmt.Errorf("provide either Derived Data Path (derived_data_path) or Xcode Project Path (project_path) Inputs")
+	}
+	if input.DerivedDataPath != "" && input.ProjectPath != "" {
+		step.logger.Warnf("Both Derived Data Path (derived_data_path) and Xcode Project Path (project_path) Inputs are provided, only derived_data_path is used, project_path is ignored")
+	}
 
-	path := filepath.Join(input.DerivedDataPath, "SourcePackages")
+	sourcePackagesPath := filepath.Join(input.DerivedDataPath, "SourcePackages")
+	if input.ProjectPath != "" {
+		var err error
+		if input.ProjectPath, err = step.pathModifier.AbsPath(input.ProjectPath); err != nil {
+			return Config{}, fmt.Errorf("failed to expand absolute project path: %w", err)
+		}
+		// project specific path already contains SourcePacages ($HOME/Library/Developer/Xcode/DerivedData/[PER_PROJECT_DERIVED_DATA]/SourcePackages)
+		if sourcePackagesPath, err = step.derivedDataPathProvider.SwiftPackagesPath(input.ProjectPath); err != nil {
+			return Config{}, fmt.Errorf("failed to get Derived Data Path: %w", err)
+		}
+	}
+
+	return Config{
+		CachePaths: sourcePackagesPath,
+		IsVerbose:  input.Verbose,
```

Comment:
Do we need to pass down the IsVerbose option? Is it used later on?

Reply:
It is passed to common cache code in line 114.